chore: update cancelGoal to handle optional id parameter

ikwilnaarhuisman opened the pull request:
The Ros1 ActionClient has the functionality that allows all goals to be canceled that were associated with the ActionClient. I thought this was very nice and useful, but missing in the Ros2 implementation of the Action / ActionClient.

Because I'm working on a project that allows for multiple users / clients to send goals and cancel those goals I don't have the correct goalID nor the means to really get it. Hence why I think this is useful.

Commit:
chore: update cancelGoal to handle optional id parameter

Commit:
chore: update cancelGoal to handle optional id parameter

Commit:
chore: update cancelGoal to handle optional id parameter

EzraBrooks reviewed (changes requested):
I don't think we should conflate "cancel a single goal" with "cancel all goals". However, this could be its own function `cancelAllGoals`, which I'd be happy to merge if you add an integration test showing it does work in ROS 2.

dependabot[bot] committed:
Build(deps-dev): Bump typescript-eslint from 8.48.0 to 8.49.0 in the eslint-and-eslint-plugins group (#1122)

Build(deps-dev): Bump typescript-eslint

Bumps the eslint-and-eslint-plugins group with 1 update: [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-eslint).


Updates `typescript-eslint` from 8.48.0 to 8.49.0
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.49.0/packages/typescript-eslint)

---
updated-dependencies:
- dependency-name: typescript-eslint
  dependency-version: 8.49.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: eslint-and-eslint-plugins
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump vite-plugin-checker from 0.11.0 to 0.12.0 (#1123)

Bumps [vite-plugin-checker](https://github.com/fi3ework/vite-plugin-checker) from 0.11.0 to 0.12.0.
- [Release notes](https://github.com/fi3ework/vite-plugin-checker/releases)
- [Commits](https://github.com/fi3ework/vite-plugin-checker/compare/vite-plugin-checker@0.11.0...vite-plugin-checker@0.12.0)

---
updated-dependencies:
- dependency-name: vite-plugin-checker
  dependency-version: 0.12.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ikwilnaarhuisman committed:
typo on sendGoal (#1126)

chore: fix typo on Action.ts

Co-authored-by: rick <rick@rick.rick>

dependabot[bot] committed:
Build(deps-dev): Bump jsdom from 27.2.0 to 27.3.0 (#1124)

Bumps [jsdom](https://github.com/jsdom/jsdom) from 27.2.0 to 27.3.0.
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/main/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/27.2.0...27.3.0)

---
updated-dependencies:
- dependency-name: jsdom
  dependency-version: 27.3.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @types/node from 24.10.0 to 25.0.2 (#1125)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 24.10.0 to 25.0.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-version: 25.0.2
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ikwilnaarhuisman committed:
Merge branch 'develop' into feat/cancel-all-goals

Commit:
chore: update cancelGoal to handle optional id parameter

Commit:
chore: update cancelGoal to handle optional id parameter

Commit:
chore: update cancelGoal to handle optional id parameter

Commit:
feat: implement integration tests for fibonacci example and cancel all goals.

Commit:
feat: implement integration tests for fibonacci example and cancel all goals.

Commit:
feat: implement integration tests for fibonacci example and cancel all goals.

Commit:
fix: skip ROS2 tests if you're not on ROS2

Commit:
fix: skip ROS2 tests if you're not on ROS2

Commit:
fix: skip ROS2 tests if you're not on ROS2

Commit:
Merge remote-tracking branch 'origin/feat/cancel-all-goals' into feat/cancel-all-goals

Commit:
Merge remote-tracking branch 'origin/feat/cancel-all-goals' into feat/cancel-all-goals

ikwilnaarhuisman commented:
closed in favour of https://github.com/RobotWebTools/roslibjs/pull/1130 because I'm not the git master I wish to be.
